src/utils/narrow.js

```diff
@@ -437,7 +438,18 @@ export const apiNarrowOfNarrow = (
       // TODO(server-2.1): just send IDs instead
       return [{ operator: 'pm-with', operand: emails.join(',') }];
     },
-    search: query => [{ operator: 'search', operand: query }],
+    // The search narrow can behave in two ways,
+    // one where there is no filters, so the user can search a message in all the streams
+    // the second where the user wants to search in a particular stream
+    // Below we are checking if the streamId is mentioned or not
+    // If it is mentioned, it will construct a narrow based on that.
+    search: (query, streamId) =>
+      streamId !== undefined
+        ? [
+            { operator: 'search', operand: query },
+            { operator: 'stream', operand: streamId },
```

Comment:
Instead of returning the stream ID here, we should handle it in the same way as in the other two places where we handle `operator: 'stream'` in this function.